Adding artifact downloads to CLI

stevenleadbeater opened the pull request.

stevenleadbeater commented:
This is failing as I can't make an issue to reference in the commit to pass the grammar checks. Can someone help me out please?

kacf commented:
> This is failing as I can't make an issue to reference in the commit to pass the grammar checks. Can someone help me out please?

Hey, you need to make sure these two are fulfilled:

* Commit must be signed off: https://github.com/mendersoftware/mender/blob/feature-c%2B%2B-client/CONTRIBUTING.md#sign-your-work (make sure the name and email matches the one in the author).
* Changelog and ticket tags must be present: https://github.com/mendersoftware/mender/blob/feature-c%2B%2B-client/CONTRIBUTING.md#structuring-your-commits

I haven't looked at the code yet.

stevenleadbeater commented:
I've done the changelog stuff and made a more detailed effort on the commit details. My commit is signed with my GPG key already. I noticed other people are adding it in their commit descriptions, I've added that too. Have I missed anything? What are these tags? I can't find an example of anyone else in the commit history

kacf commented:
You haven't pushed any new changes since I made my last comment, so I can't see exactly what you did.

The signoff is sort of a signature, not cryptographic, but like one would do on a contract. It's signals that you are vouching for the code you have submitted, and you are allowed to publish it (i.e it's either your copyright, or someone you legally represent).

f989fa1d19b1ff7963bf9ddc43f171575ddcd970 is an example of a commit where you can see the tags.

stevenleadbeater commented:
@kacf  sorry I assumed the github PR comment was enough, I've updated the commit message. I still don't follow what the "Changelog: Title" is for but I added it anyway, assuming its an instruction to your CI to use the title (feat: <title> section) of the commit message in the changelog

kacf commented on client/deployments/client.go:
```diff
@@ -390,3 +396,202 @@ func (c *Client) DeleteArtifact(
+	} else {
+		n, err = io.Copy(file, source)
+	}
+	if err != nil {
+		log.Verbf("wrote: %d\n", n)
+		return err
+	}
+	log.Verbf("wrote: %d\n", n)
```
Nitpick: Could move the `log` before the `if`, since you do the same in both cases.

stevenleadbeater replied:
done in both places :+1:

kacf reviewed (changes requested):
@stevenleadbeater: Thanks, the commit text looks good now.

I looked at the code also, nice work! I'm requesting one tiny nitpick/cleanup, but other than that it looks great to me.

kacf commented:
Thanks, @stevenleadbeater, I fixed a couple of minor style issues.

It looks like the test doesn't agree with us though. I haven't looked at the details yet, but the test result is [here](https://gitlab.com/Northern.tech/Mender/mender-cli/-/jobs/4838935287) (the relevant failure is about one screenful up from the bottom). Does it ring a bell for you?

kacf commented:
Merged! Thank you very much, @stevenleadbeater!
